.github/workflows/check-changelog.yml

```diff
@@ -0,0 +1,52 @@
+name: Check Changelog
+# This check makes sure that the changelog is properly updated
+# when a PR introduces a change in a test file.
+# To bypass this check, label the PR with "No Changelog Needed".
+on:
+  pull_request:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'No Changelog Needed') == 0 }}
+    steps:
+      - name: Get PR number and milestone
+        run: |
+          echo "PR_NUMBER=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+          echo "MILESTONE_TITLE=${{ github.event.pull_request.milestone.title }} >> $GITHUB_ENV
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: '0'
+      - name: Check the changelog
+        run: |
+          set -xe
+          changed_files=$(git diff --name-only origin/master)
+          # Changelog should be updated only if tests have been modified
+          if [[ "$changed_files" =~ tests ]]
```

Comment:
I'd still suggest to exit early here if no tests are changed: this saves one nested block, which helps readability (especially in bash ;) )

Reply:
I've seen your previous comment and and replied there: I'm sorry but I can't understand what you are suggesting.
Should I add an `else` with `exit 0`? The end of this `if` is the end of the script, I guess if the check is False there is nothing else to be executed...

Reply:
Sorry I missed your reply up there. Basically I'm suggesting to do

```bash
if [[ ! "$changed_files" =~ tests ]]  # added the "!" here
then
	exit 0
fi  # <- move very last fi here

# continue as before but everything indented right
```